src/ui/repo-tree-model.cpp

```diff
@@ -61,8 +61,11 @@ RepoTreeModel::RepoTreeModel(QObject *parent)
 }
 RepoTreeModel::~RepoTreeModel()
 {
-    if (item(kIndexOfVirtualReposCategory) != virtual_repos_category_)
-        delete virtual_repos_category_;
+    // if we miss it, just add it and
+    // let dtor of repotreemodel to delete it automatically
+    if (item(kIndexOfVirtualReposCategory) != virtual_repos_category_) {
+        insertRow(kIndexOfVirtualReposCategory, virtual_repos_category_);
+    }
```

Comment:
I have reverted this part since I am not sure if the solution provided solves the memory leaks. If we come back again, make sure we eliminate this leak in the next patch.